### PR TITLE
test: Demonstrate e2e tests for private IP databases and GKE clusters.

### PR DESCRIPTION
**Pull request by hessjcg**
Demonstrate that the Private IP operator flag correctly allows the proxy to connect to a private IP database.

### CODE REVIEW

File: infra/resources/main.tf [hunk 1]
```diff
@@ -27,6 +27,11 @@ terraform {
 provider "google" {
 }
 
+provider "google-beta" {
```
Comment (enocom) [inline infra/resources/main.tf:30]: What requires google beta here?

File: tests/e2e_test.go [hunk 2]
```diff
@@ -487,3 +487,111 @@ func TestUpdateWorkloadOnDelete(t *testing.T) {
...
+	)
+
+	tests := []struct {
+		name        string
+		c           *testhelpers.TestCaseClient
+		podTemplate corev1.PodTemplateSpec
+		allOrAny    string
+	}{
+		{
+			name:        "postgres",
```
Comment (enocom) [inline tests/e2e_test.go:509]: Are you planning on adding additional test cases here?

File: infra/resources/main.tf (hunk 1 above)
Comment (hessjcg) [reply infra/resources/main.tf:30]: Only the `google-beta` provider for the `google_sql_database_instance` for supports private networking. See [google-sql-database-instance Terraform Doc](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#private-ip-instance)

Comment (enocom) [reply infra/resources/main.tf:30]: In the example, they pull this provider setting under the resource that actually needs it. Shall we do that here?

**Review by enocom — APPROVED** [review]
:ship:

File: infra/resources/main.tf (hunk 1 above)
Comment (hessjcg) [reply infra/resources/main.tf:30]: I want to keep the providers all together `main.tf` because it is easier to manage.

**Comment by hessjcg** [comment]
I want to keep all the provider declarations together in `main.tf` so that they are easier to manage.

**Pull request merged**